core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -165,6 +169,25 @@ public String getName() {
     }
 
     @Override
+    protected boolean updateOutputSockets() {
```

Comment:
1. If you have multiple camera sources then that will throttle the pipeline to the slowest one.
The way that I implemented we only run the pipeline when any of the sources/steps have new values.

2. Imagine in an extreme case where you have an IPCamera that provides images at 1 frame per 5 seconds. You want to be able to change pipeline values and see those changes without having the pipeline be blocked on one source.

3. Also, what happens if a source crashes? This can cause the `grab` method to hang for some indeterminate length of time. That would halt the entire pipeline until `grab` returns.